src/Common/CSharp/Analysis/UseConstantInsteadOfFieldAnalysis.cs

```diff
@@ -82,6 +83,18 @@ internal static class UseConstantInsteadOfFieldAnalysis
 
             if (!semanticModel.HasConstantValue(value, cancellationToken))
                 return false;
+
+            // Changing a static field to a constant changes the meaning of that symbol in the decorators.
+            foreach (IdentifierNameSyntax identifierName in value.DescendantNodesAndSelf().OfType<IdentifierNameSyntax>())
```

Comment:
I'd be slightly more effective if cast to IdentifierNameSyntax would be inside foreach (as first `if` condition)